src/iceberg/catalog/rest/rest_catalog.cc

```diff
@@ -44,53 +47,85 @@ namespace iceberg::rest {
 
 namespace {
 
-// Fetch server config and merge it with client config
-Result<std::unique_ptr<RestCatalogProperties>> FetchConfig(
-    const ResourcePaths& paths, const RestCatalogProperties& config) {
-  ICEBERG_ASSIGN_OR_RAISE(auto config_endpoint, paths.Config());
-  HttpClient client(config.ExtractHeaders());
+/// \brief Get the default set of endpoints for backwards compatibility according to the
+/// iceberg rest spec.
+std::unordered_set<Endpoint, EndpointHash> GetDefaultEndpoints() {
+  return {
+      Endpoint::ListNamespaces(),  Endpoint::GetNamespaceProperties(),
+      Endpoint::CreateNamespace(), Endpoint::UpdateNamespace(),
+      Endpoint::DropNamespace(),   Endpoint::ListTables(),
+      Endpoint::LoadTable(),       Endpoint::CreateTable(),
+      Endpoint::UpdateTable(),     Endpoint::DeleteTable(),
+      Endpoint::RenameTable(),     Endpoint::RegisterTable(),
+      Endpoint::ReportMetrics(),   Endpoint::CommitTransaction(),
+  };
+}
+
+/// \brief Fetch server config and merge it with client config
+Result<CatalogConfig> FetchServerConfig(const ResourcePaths& paths,
+                                        const RestCatalogProperties& current_config) {
+  ICEBERG_ASSIGN_OR_RAISE(auto config_path, paths.Config());
+  HttpClient client(current_config.ExtractHeaders());
   ICEBERG_ASSIGN_OR_RAISE(const auto response,
-                          client.Get(config_endpoint, /*params=*/{}, /*headers=*/{},
+                          client.Get(config_path, /*params=*/{}, /*headers=*/{},
                                      *DefaultErrorHandler::Instance()));
   ICEBERG_ASSIGN_OR_RAISE(auto json, FromJsonString(response.body()));
-  ICEBERG_ASSIGN_OR_RAISE(auto server_config, CatalogConfigFromJson(json));
-
-  // Merge priorities: server overrides > client config > server defaults
-  return RestCatalogProperties::FromMap(
-      MergeConfigs(server_config.overrides, config.configs(), server_config.defaults));
+  return CatalogConfigFromJson(json);
 }
 
 }  // namespace
 
 RestCatalog::~RestCatalog() = default;
 
 Result<std::unique_ptr<RestCatalog>> RestCatalog::Make(
-    const RestCatalogProperties& config) {
-  ICEBERG_ASSIGN_OR_RAISE(auto uri, config.Uri());
+    const RestCatalogProperties& initial_config) {
+  ICEBERG_ASSIGN_OR_RAISE(auto uri, initial_config.Uri());
```

Comment:
```suggestion
    const RestCatalogProperties& config) {
  ICEBERG_ASSIGN_OR_RAISE(auto uri, config.Uri());
```

This is inconsistent with the function declaration. I think its original name is better.